frontend/src/metabase/query_builder/components/view/PreviewQueryModal/PreviewQueryModal.tsx

```diff
@@ -29,72 +14,24 @@ const PreviewQueryModal = ({
   question,
   onClose,
 }: PreviewQueryModalProps): JSX.Element => {
-  const { data, error, isLoading } = useNativeQuery(question);
-
-  const queryText = useMemo(() => {
-    const engine = question.database()?.engine;
-    return data ? formatNativeQuery(data.query, engine) : undefined;
-  }, [question, data]);
-
-  const errorText = useMemo(() => {
-    return error ? getErrorMessage(error) : undefined;
-  }, [error]);
+  const { query, error, isLoading } = useNativeQuery(question);
```

Comment:
Due to how the modal is being unmounted and mounted over and over again every time we click `View the SQL` button. This hook will keep calling the API over and over again. But in the previous version (stats), we only call the API once.

I think the code is simpler but don't think we're ok with calling the API too many times.

#### Stats
<img width="497" alt="image" src="https://user-images.githubusercontent.com/1937582/206693390-e7a97005-3154-416e-8022-14be3acc3a95.png">

#### This PR
<img width="506" alt="image" src="https://user-images.githubusercontent.com/1937582/206693440-280c7c32-182c-4f7d-bfd5-f70fad265107.png">

Reply:
I'm 99% sure it's ok to call it multiple times here as this is not something you'd use often.

Reply:
If there are no other blockers I could merge this PR first and try to cache `dataset/native` results in another PR